File: forestrymanagementsystemspringrest/src/main/java/com/capgemini/forestrymanagementsystemspringrest/service/CustomerService.java
package com.capgemini.forestrymanagementsystemspringrest.service;

import java.util.List;

import com.capgemini.forestrymanagementsystemspringrest.dto.CustomerBean;

public interface CustomerService {
	public boolean addCustomer(CustomerBean customer);
	public boolean deleteCustomer(int customerid);
	public List<CustomerBean> getAllCustomer();
	public boolean modifyCustomer(int customerid,CustomerBean customer );
	public CustomerBean getCustomer(int customerid);


}

File: SpringRestUI/src/app/customer.service.ts
import { Injectable } from '@angular/core';
import {HttpClient, HttpHeaders} from '@angular/common/http';
import { Observable } from 'rxjs';

@Injectable({
  providedIn: 'root'
})
export class CustomerService {
  api = 'http://localhost:8080';
  selectedCustomerToUpdate;
  constructor(private http: HttpClient) { }

  postData(customers): Observable<any> {
    return this.http.post<any>(
      `http://localhost:8080/springrest/addcustomer`,
      customers);
   }

    getData(): Observable<any> {
    return this.http.get<any>(`http://localhost:8080/springrest/getAllcustomer`);
    }
    deleteData(customers: any): Observable<any> {
      return this.http.delete<any>(`http://localhost:8080/springrest/deletecustomer/${customers.customerid}`);
    }
    updateData(customers): Observable<any> {
      return this.http.put<any>(
        `http://localhost:8080/springrest/modifycustomer`,
        customers
      );
    }
}
